queries.sql

```diff
@@ -56,3 +56,68 @@ SELECT o.full_name AS Owner_name, a.name AS Digimon FROM animals a INNER JOIN ow
 SELECT o.full_name AS owner_name, a.name AS animal_name FROM animals a INNER JOIN owners o ON a.owner_id = o.id WHERE o.full_name = 'Dean Winchester' AND a.escape_attempts = 0;
 SELECT o.full_name AS owner_name, COUNT(a.name) AS nb_animal FROM animals a INNER JOIN owners o ON a.owner_id = o.id GROUP BY o.full_name ORDER BY nb_animal DESC LIMIT 1;
 
+
+SELECT a.name, v.name, vs.date_of_visit
+FROM animals a
+INNER JOIN visits vs ON a.id = vs.animal_id
+INNER JOIN vets v ON v.id = vs.vet_id
+WHERE v.name = 'William Tatcher'
+ORDER BY vs.date_of_visit DESC LIMIT 1;
+
+SELECT v.name, COUNT(a.name) AS nb_animal
+FROM animals a
+INNER JOIN visits vs ON a.id = vs.animal_id
+INNER JOIN vets v ON v.id = vs.vet_id
+WHERE v.name = 'Stephanie Mendez' GROUP BY v.name;
+
+SELECT v.name, s.name AS specialized
+FROM vets v
+LEFT JOIN specializations sp ON v.id = sp.vet_id
+LEFT JOIN species s ON s.id = sp.specie_id;
+
+SELECT v.name, a.name AS animal_name, vs.date_of_visit
+FROM animals a
+INNER JOIN visits vs ON a.id = vs.animal_id
+INNER JOIN vets v ON v.id = vs.vet_id
+WHERE vs.date_of_visit BETWEEN '2020-04-01' AND '2020-08-30' AND v.name = 'Stephanie Mendez';
+
+SELECT a.name, COUNT(vs.animal_id) AS nb_visit
+FROM animals a
+INNER JOIN visits vs ON a.id = vs.animal_id
+GROUP BY a.name ORDER BY COUNT(vs.animal_id) DESC LIMIT 1;
+
+SELECT v.name AS vet, a.name, vs.date_of_visit
+FROM visits vs
+INNER JOIN animals a ON vs.animal_id = a.id
+INNER JOIN vets v ON v.id = vs.vet_id
+ORDER BY vs.date_of_visit ASC LIMIT 1;
+
+SELECT v.name AS vet_name, s.name AS specialized_in, a.name AS animal_name,
+s.name AS species, a.date_of_birth, a.neutered, a.weight_kg,
+vs.date_of_visit AS recent_visit
+FROM animals a
+INNER JOIN visits vs ON a.id = vs.animal_id
+INNER JOIN vets v ON v.id = vs.vet_id
+INNER JOIN specializations sp ON sp.vet_id = v.id
+INNER JOIN species s ON s.id = sp.specie_id
+GROUP BY v.name, a.name, s.name, a.date_of_birth, a.neutered, a.weight_kg, vs.date_of_visit
+ORDER BY vs.date_of_visit DESC LIMIT 1;
+
+SELECT vt.name AS vet, COUNT(*) AS visits_with_no_specialization
+FROM visits v
+JOIN animals a ON v.animal_id = a.id
+JOIN species s ON a.species_id = s.id
+JOIN vets vt ON v.vet_id = vt.id
+LEFT JOIN specializations sp ON v.vet_id = sp.vet_id AND s.id = sp.specie_id
+WHERE sp.vet_id IS NULL GROUP BY vt.name;
+
+SELECT vt.name AS vet_name, s.name AS species_specialisation
+FROM visits v
+JOIN vets vt ON v.vet_id = vt.id
+JOIN animals a ON v.animal_id = a.id
+JOIN species s ON a.species_id = s.id
+WHERE vt.name = 'Maisy Smith'
+GROUP BY vt.name, s.name
+ORDER BY COUNT(v.animal_id)
+DESC LIMIT 1;
```

Comment:
- [ ] Great job working on the functionality so far but I think it would be great if you fixed this query here because if we want to answer this question `What specialty should Maisy Smith consider getting? Look for the species she gets the most.` so she gets the most `Boarmon` not `Digmon` as shown in your screenshot so kindly instead of counting it you can add this condition instead  ✅🚀💯


```sql
  WHERE vet.name = 'Maisy Smith'
  GROUP BY species.name
```